extensions/extb_canjoin.c

```diff
@@ -63,7 +64,51 @@ static int eb_canjoin(const char *data, struct Client *client_p,
 		return EXTBAN_INVALID;
 #endif
 	recurse = 1;
-	ret = is_banned(chptr2, client_p, NULL, NULL, NULL) != 0 ? EXTBAN_MATCH : EXTBAN_NOMATCH;
+	/* Check both bans and quiets in the target channel for $j extban */
+	const char *forward = NULL;
+	struct matchset ms;
+	matchset_for_client(client_p, &ms);
+	int banned_result = is_banned(chptr2, client_p, NULL, &ms, &forward);
+	if (banned_result != 0)
+	{
+		/* If the ban has a forward, check if user would be banned in the forward channel */
+		if (forward != NULL)
+		{
+			struct Channel *forward_chptr = find_channel(forward);
+			if (forward_chptr != NULL && forward_chptr != chptr && forward_chptr != chptr2)
+			{
+				/* Check if user is banned in forward channel (but don't recurse $j checks) */
+				recurse = 0;
```

Comment:
`recurse` is used to track whether we are checking bans from inside a `$j`. Unsetting it here will cause the opposite of what's intended and allow for (potentially infinite) recursion.